src/p2p/peer.rs

```diff
@@ -53,28 +54,21 @@ impl Peer {
         }
     }
 
-    async fn handle_consensus_message(&mut self, msg: ConsensusNetMessage) -> Result<(), Error> {
-        send_net_message(&mut self.stream, NetMessage::ConsensusMessage(msg)).await
-    }
-
-    async fn handle_mempool_message(&mut self, msg: MempoolNetMessage) -> Result<(), Error> {
-        send_net_message(&mut self.stream, NetMessage::MempoolMessage(msg)).await
-    }
-
-    async fn handle_broadcast_message(&mut self, msg: Broadcast) -> Result<(), Error> {
-        if msg.peer_id == self.id {
+    async fn handle_send_message(&mut self, msg: SendMessage) -> Result<(), Error> {
+        if msg.peer_id != self.id {
```

Comment:
C'est pas plutot 
```suggestion
        if msg.peer_id == self.id {
```
Pour ne rien faire si le node l'envoit à lui-même ?